spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -80,6 +80,8 @@
 
 	private volatile Runnable stopCallback;
 
+	private volatile boolean readFromLeft = false;
```

Comment:
Since we are basically creating a "queue" abstraction over a Redis LIST collection type, perhaps it would be better to call this property `fifo` - with a default of `true`; set to `false` for LIFO (left pop is an implementation detail in this case).